### ISSUE TITLE
Enhance the possible heading warning to not flag elements that have been made heading with ARIA

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1685]
Code like this:
`<p class="cky-title" data-cky-tag="title" aria-level="2" role="heading" style="color: #212121;">We value your privacy</p>`

Elements that have  `role="heading"` and `aria-level` attributes are headings and should not flag a possible heading issue, but they do currently:

<img width="1043" height="133" alt="Image" src="https://github.com/user-attachments/assets/58a13321-add3-4ddb-ba99-fe11e067acc9" />

**Comment by pattonwebz** [comment]
@amberhinds do you know if we also have to update our skipped header level checks as well to account for this pattern?

edit: I had tests confirm that the rule we check for skipped headings with does account for when headings are created via a role with an aria-level - as well as heading tags. So no need to make any changes to the incorrect heading order checks.

**Comment by amberhinds** [comment]
Yeah, I think @SteveJonesDev  already accounted for ARIA headings in skipped heading levels. @pattonwebz you might check to see if the missing heading check accounts for ARIA headings.

**Comment by pattonwebz** [comment]
I checked the skip rules, and they do account for the role properly. I will check the missing header check as well shortly.

@amberhinds I have a PR up that adds your request to the possible heading rule. You can [test it on playground](https://playground.wordpress.net/?blueprint-url=data%3Aapplication%2Fjson%3Bbase64%2CeyJsYW5kaW5nUGFnZSI6Ii93cC1hZG1pbi9wbHVnaW5zLnBocCIsInN0ZXBzIjpbeyJzdGVwIjoibG9naW4ifSx7InN0ZXAiOiJpbnN0YWxsUGx1Z2luIiwicGx1Z2luWmlwRmlsZSI6eyJyZXNvdXJjZSI6InVybCIsInVybCI6Imh0dHBzOi8vbmlnaHRseS5saW5rL2VxdWFsaXplZGlnaXRhbC9hY2Nlc3NpYmlsaXR5LWNoZWNrZXIvYWN0aW9ucy9ydW5zLzI1NDYyOTc0NjIwL2FjY2Vzc2liaWxpdHktY2hlY2tlci0xLjQwLjAtMTY4Ni02Nzg0OGRlZC56aXAifX0seyJzdGVwIjoiYWN0aXZhdGVQbHVnaW4iLCJwbHVnaW5QYXRoIjoiYWNjZXNzaWJpbGl0eS1jaGVja2VyL2FjY2Vzc2liaWxpdHktY2hlY2tlci5waHAifV19). I tested with this markup:

```
<h2>A real h2 tag</h2>
<p style="font-weight: 800">A bold paragraph</p>
<p style="font-weight: 800" role="heading">A bold paragraph with header role</p>
<p style="font-weight: 800" role="heading" aria-level="3">A bold paragraph with header role and aria-level 3</p>
<p role="heading" aria-level="5">A paragraph with header role and aria-level 5 - skipped level</p>
```

**Comment by amberhinds** [comment]
This looks good.

**Comment by pattonwebz** [comment]
We confirmed this morning that the missing headings rule does have an accordance for role="heading" so no need to update that, it was just this rule that was expecting pure header tags and not account for roles.
